client/my-sites/post-type-list/index.jsx

```diff
@@ -1,60 +1,77 @@
 /**
  * External dependencies
  */
-import React, { PropTypes } from 'react';
+import React, { Component, PropTypes } from 'react';
 import { connect } from 'react-redux';
+import isEqual from 'lodash/isEqual';
 
 /**
  * Internal dependencies
  */
 import QueryPosts from 'components/data/query-posts';
 import { getSelectedSiteId } from 'state/ui/selectors';
-import {
-	isRequestingSitePostsForQuery,
-	getSitePostsForQueryIgnoringPage
-} from 'state/posts/selectors';
-import PostTypeListPost from './post';
-import PostTypeListPostPlaceholder from './post-placeholder';
-import PostTypeListEmptyContent from './empty-content';
-
-function PostTypeList( { query, siteId, posts, requesting } ) {
-	return (
-		<div className="post-type-list">
-			<QueryPosts
-				siteId={ siteId }
-				query={ query } />
-			{ posts && ! posts.length && ! requesting && (
-				<PostTypeListEmptyContent
-					type={ query.type }
-					status={ query.status } />
-			) }
-			<ul className="post-type-list__posts">
-				{ requesting && (
-					<li><PostTypeListPostPlaceholder /></li>
-				) }
-				{ posts && posts.map( ( post ) => (
-					<li key={ post.global_ID }>
-						<PostTypeListPost globalId={ post.global_ID } />
-					</li>
-				) ) }
-			</ul>
-		</div>
-	);
+import PostTypeListList from './list';
+
+/**
+ * Constants
+ */
+const POSTS_PER_PAGE = 10;
+
+class PostTypeList extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.boundSetNextPage = this.setNextPage.bind( this );
+
+		this.state = PostTypeList.initialState;
+	}
+
+	getQuery() {
+		return Object.assign( {}, this.props.query, {
+			page: this.state.page,
+			number: POSTS_PER_PAGE
+		} );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( ! isEqual( nextProps.query, this.props.query ) ) {
+			this.setState( PostTypeList.initialState );
+		}
+	}
+
+	setNextPage() {
+		this.setState( {
+			page: this.state.page + 1
+		} );
+	}
+
+	render() {
+		const query = this.getQuery();
+
+		return (
+			<div className="post-type-list">
+				<QueryPosts
+					siteId={ this.props.siteId }
+					query={ query } />
+				<PostTypeListList
```

Comment:
>ListList? <a href="https://en.wikipedia.org/wiki/Franz_Liszt">Liszt</a>!

I suppose we can be grateful it's not a list of lists? `<PostTypeListListList />` !